test/EndToEndTests/Tests/Client/Build.Desktop/DeltaTests/DeltaTests.cs

```diff
@@ -161,42 +162,47 @@ public void RequestDeltaLink_Expanded()
                 {
                     using (var messageReader = new ODataMessageReader(responseMessage, readerSettings, Model))
                     {
-                        var deltaReader = messageReader.CreateODataDeltaReader(customersSet, customerType);
+                        var deltaReader = messageReader.CreateODataDeltaResourceSetReader(customersSet, customerType);
 
                         while (deltaReader.Read())
                         {
-                            if (deltaReader.State == ODataDeltaReaderState.DeltaResourceSetEnd)
+                            if (deltaReader.State == ODataReaderState.DeltaResourceSetEnd)
                             {
                                 Assert.NotNull(deltaReader.Item as ODataDeltaResourceSet);
                             }
-                            else if (deltaReader.State == ODataDeltaReaderState.DeltaResourceEnd)
+                            else if (deltaReader.State == ODataReaderState.ResourceEnd)
                             {
                                 Assert.NotNull(deltaReader.Item as ODataResource);
                             }
-                            else if (deltaReader.State == ODataDeltaReaderState.NestedResource)
+                            else if (deltaReader.State == ODataReaderState.NestedResourceInfoStart)
                             {
-                                switch (deltaReader.SubState)
-                                {
-                                    case ODataReaderState.Start:
-                                        Assert.NotNull(deltaReader.Item as ODataNestedResourceInfo);
-                                        break;
-                                    case ODataReaderState.ResourceSetEnd:
-                                        Assert.NotNull(deltaReader.Item as ODataResourceSet);
-                                        break;
-                                    case ODataReaderState.ResourceEnd:
-                                        Assert.NotNull(deltaReader.Item as ODataResource);
-                                        break;
-                                    case ODataReaderState.NestedResourceInfoEnd:
-                                        Assert.NotNull(deltaReader.Item as ODataNestedResourceInfo);
-                                        break;
-                                    case ODataReaderState.Completed:
-                                        Assert.NotNull(deltaReader.Item);
-                                        break;
-                                }
+
+                            }
+                            else if (deltaReader.State == ODataReaderState.NestedResourceInfoEnd)
+                            {
+                                Assert.NotNull(deltaReader.Item as ODataNestedResourceInfo);
+                                //switch (deltaReader.SubState)

```

Comment:
Please delete unused code, you can always retrieve them from Git.

Reply:
@wachugamaina I'll replace the commented lines with better assert statements. I noticed that tests have insufficient assert statements. Checking that the item is not null doesn't feel sufficient enough to me